File: CreationalPatterns/Prototype/TypeScript/Circle.ts
import { Shape } from './Shape';

export class Circle extends Shape {
    radius: number = 0;

    constructor(source?: Circle) {
        super(source);
        if (source) {
            this.radius = source.radius;
        }
    }

    clone(): Circle {
        return new Circle(this);
    }

    toString(): string {
        return `Circle: x=${this.x}, y=${this.y}, color=${this.color}, radius=${this.radius}`;
    }
}

File: CreationalPatterns/Prototype/CSharp/Circle.cs
namespace Prototype
{
    public class Circle : Shape
    {
        public int Radius { get; set; }

        public Circle() { }
        public Circle(Circle source) : base(source)
        {
            Radius = source.Radius;
        }

        public override Circle Clone()
        {
            return new Circle(this);
        }

        public override string ToString()
        {
            return $"Circle: X={X}, Y={Y}, Color={Color}, Radius={Radius}";
        }
    }
}

